junit-jupiter-engine/src/test/java/org/junit/jupiter/engine/AtypicalJvmMethodNameTests.java

```diff
@@ -31,15 +32,18 @@
 
 	@Test
 	void kotlinTestWithMethodNameContainingSpecialCharacters() {
-		ExecutionEventRecorder eventRecorder = executeTestsForClass(ArbitraryNamingKotlinTestCase.class);
-		assertThat(eventRecorder.getTestFinishedCount()).isEqualTo(2);
+		ExecutionsResult executionsResult = executeTestsForClass(
+			ArbitraryNamingKotlinTestCase.class).getExecutionsResult();
+		assertThat(executionsResult.getTestsStartedCount()).isEqualTo(2);
```

Comment:
From `.getTestFinishedCount()` to `.getTestsStartedCount()`. Although the numbers reported may be equal, the meaning of _finished_ and _started_ are not interchangeable.

Please provide the old method in `ExecutionsResult` with exactly the same meaning and spelling. This also means to leave the `s` from `Test_s_` out for a future commit/PR.

Reply:
This makes sense but I would like a bit more clarification on the renaming.

First, I can definitely switch the meaning to reference the same call to the `executionsByTerminationInfo()` as the `*Finished*()` methods, and change the doc accordingly.

Second, in terms of renaming, I wasn't sure where to draw the line on pluralization of the concepts returned by the API. The core concepts being:
- Event
- TestEvent
- ContainerEvent
- Test

It seemed more straightforward to keep a single type of pluralization across all API calls for each concept, with the top level non-filtering calls being the most obvious to pluralize (ie. `getEvents()` / `getTests()` / etc).

However if I am understanding you correctly, applying the rename you are requesting then we would be left with some API calls being pluralized while others are not (of the same granularity of filter), **causing the API method names for `ExecutionsResult` to not be symmetrical across the concepts.**

Using your example of changing `getTestsFinishedCount()` to `getTestFinishedCount()`, would that mean that I leave the top level non-filtering call to still be plural (`getTests()`), while also not changing the similar granularity filter calls for the other concepts? In other words, should I also change the pluralization of all other concept calls (`getTestEventsFinishedCount()` -> `getTestEventFinishedCount()` / `getContainerEventsFinishedCount()` -> `getContainerEventFinishedCount()` / etc) in order to maintain complete API symmetry?

Apologies in advance if I completely misunderstood this.

Reply:
Regarding the renaming and pluralization -- I'm totally in favor to have a consistent naming pattern across the *testkit* API. In a _subsequent_ commit or even a follow-up PR.

Let's first focus on introducing the new subproject by moving logic from the test-scope (like `org.junit.platform.engine.test` packages) to the main-scope: `org.junit.platform.testkit`. That's what I wanted (but failed?) to express some weeks ago with: _In general, I'd like to see more "package-only" refactorings. Like only replacing `engine.test` with `testkit` to keep the scope of the PR concise._

Before releasing *testkit* into the wild, we need to consider your thoughts and work already done to present a nice, concise, consistent and clear API.

One step after the other.

Reply:
Thanks for clarifying and putting up with my questions @sormuras!  Sorry I didn't understand that at first.  That makes complete sense now 👍 

I will action on these changes within the next couple hours, thanks again for your feedback.